site/content/en/docs/technical-outcomes/simulation.md

```diff
@@ -0,0 +1,170 @@
+---
+title: Simulation Testing
+---
+
+# Simulation Testing with KWOK
+
+{{< hint "info" >}}
+
+This document walks you through the technical outcome of using KWOK for simulation testing.
+
+{{< /hint >}}
+
+This article describes the technical outcome of using [KWOK] for simulation testing in your environment.
+It goes over the key simulation features of using [KWOK], and then provides a step-by-step guide to creating
+various scheduling scenarios using [KWOK].
+
+## Key Simulation Features of KWOK
+
+1. **Lifecycle configuration:** The pod, node, and node lease can be simulated.
+
+    - **Pod:** Status information like Pending, ContainerCreating, Running, and Terminating can be simulated.
+    - **Node:** Status information like NodeReady, and Ready states can be simulated.
+
+2. **Kubelet server**
+
+    - **Metrics:** Using the [Kubernetes Metric Server](https://kwok.sigs.k8s.io/docs/examples/metrics-server/), or [Prometheus](https://kwok.sigs.k8s.io/docs/examples/prometheus/), metrics can be simulated in KWOK.
+    - **Exec:** Commands can be simulated to be [executed](https://kwok.sigs.k8s.io/docs/user/exec-configuration/) inside a container running on a single fake pod or multiple pods.
+    - **Log:** Container [logs](https://kwok.sigs.k8s.io/docs/user/logs-configuration/) can be simulated in a single pod or multiple pods.
+    - **Attach:** The input/output streams from the main process running in a container can be [attached](https://kwok.sigs.k8s.io/docs/user/attach-configuration/#attach-configuration) to KWOK.
+    This can be simulated in a single pod or multiple pods.
+    - **PortForward:** [Port forwarding](https://kwok.sigs.k8s.io/docs/user/port-forward-configuration/#clusterportforward), which establishes a direct connection to a container port, can be simulated
+    for a single pod or multiple pods.
+
+## Technical outcomes
+
+### Scheduler simulation
+
+[KWOK] can be used to create fake nodes and pods in a simulated cluster.
+The cluster can be configured with scheduling policies that meet your scheduler's requirements.
+Two scenarios will be used to describe this.
+
+#### Scenario 1: Scheduling pods with resource requests and limits
+
+##### Prerequisites
+
+- [KWOK] must be installed on the machine. See [installation](https://kwok.sigs.k8s.io/docs/user/installation/).
+
+##### Create cluster
+
+```bash
+kwokctl create cluster
+```
+
+##### View clusters
+
+- This ensures that the cluster was created successfully.
+
+```bash
+kwokctl get clusters
+```
+
+##### Deploy nodes
+
+Below are the node resource specifications:
+
+- Nodes: 2 worker nodes in the cluster
+  - Node 1: 2 CPUs, 2 GB RAM
+  - Node 2: 4 CPUs, 4 GB RAM
+<!-- I am going to change each bash command to reference the file via a url. -->
+```bash
+kubectl apply -f node.yaml
```

Comment:
Hi @wzshiming, how can we specify the URL to the `node.yaml` file and every other referenced file?

I know the URL to the demo path on the main branch is "https://github.com/kubernetes-sigs/kwok/tree/main/demo"

Reply:
Lovely, I will.